narwhal/consensus/src/tests/consensus_tests.rs

```diff
@@ -386,6 +390,70 @@ async fn test_leader_swap_table() {
     }
 }
 
+#[tokio::test]
+async fn test_leader_schedule() {
+    // GIVEN
+    let fixture = CommitteeFixture::builder().build();
+    let committee = fixture.committee();
+
+    // the authority ids
+    let authority_ids: Vec<AuthorityIdentifier> = fixture.authorities().map(|a| a.id()).collect();
+
+    // Create a leader schedule with a default swap table, so no authority will be swapped.
+    let schedule = LeaderSchedule::new(committee.clone(), LeaderSwapTable::default());
+
+    // Call the leader for round 2. It should give us the validator of position 0
+    let original_leader = authority_ids[0];
```

Comment:
Can we calculate leader with the committee? In case how tests calculate leaders are changed.

Reply:
hhm not sure what you mean here. Could you please clarify?

Reply:
I mean instead of assuming `authority_ids[0]` is the leader, is it possible to calculate the leader with a library method? Otherwise it would be pretty perplexing if the test needs to be fixed later.

Reply:
Well that's what we are testing here though - the line `407` is practically deriving the leader via the library. To ensure the validity of the result I _have_ to assert against an expected value which is the `authority_ids[0]`. Unless I am still missing the point you are making